src/ServiceInsight/Framework/Settings/ApplicationConfiguration.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace ServiceInsight.Framework.Settings
+{
+    using System;
+    using System.Configuration;
+
+    public static class ApplicationConfiguration
+    {
+        public static bool SkipCertificateValidation { get; private set; }
+
+        public static void Initialize()
+        {
+            SkipCertificateValidation = GetSkipCertificateValidation();
+        }
+
+        private static bool GetSkipCertificateValidation()
+        {
+            try
+            {
+                return Convert.ToBoolean(ConfigurationManager.AppSettings["SkipCertificateValidation"] ?? "false");
```

Comment:
Perhaps I'm getting too far into an edge case, but could there be a situation where some other exception is thrown, and we disguise it as the value not being a valid Boolean? E.g. a problem accessing the config file itself? To be safe, should we only try...catch `Convert.ToBoolean()` and not `ConfigurationManager.AppSettings[]`?

Reply:
@adamralph in which case, the error message + inner exception should be clear enough to find that out, right?

Reply:
Exception message doesn't match. If the value is missing the default will be false. The `is either missing or` doesn't make sense.

Reply:
@WilliamBZA I've already rolled that change back though. See the revert commit below.